src/mongoc/mongoc-uri.c

```diff
@@ -468,6 +468,41 @@ mongoc_uri_parse_tags (mongoc_uri_t *uri, /* IN */
    bson_destroy(&b);
 }
 
+bool
+mongoc_uri_option_is_int32 (const char *key)
+{
+   return !strcasecmp(key, "connecttimeoutms") ||
+       !strcasecmp(key, "heartbeatfrequencyms") ||
+       !strcasecmp(key, "serverselectiontimeoutms") ||
+       !strcasecmp(key, "socketcheckintervalms") ||
+       !strcasecmp(key, "sockettimeoutms") ||
+       !strcasecmp(key, "maxpoolsize") ||
+       !strcasecmp(key, "minpoolsize") ||
+       !strcasecmp(key, "maxidletimems") ||
+       !strcasecmp(key, "waitqueuemultiple") ||
+       !strcasecmp(key, "waitqueuetimeoutms") ||
+       !strcasecmp(key, "wtimeoutms");
+}
+
+bool
+mongoc_uri_option_is_bool (const char *key)
+{
+   return !strcasecmp(key, "canonicalizeHostname") ||
+              !strcasecmp(key, "journal") ||
+              !strcasecmp(key, "safe") ||
+              !strcasecmp(key, "slaveok") ||
+              !strcasecmp(key, "ssl");
+}
+
+bool
+mongoc_uri_option_is_utf8 (const char *key)
+{
+   return !strcasecmp(key, "authmechanism") ||
+              !strcasecmp(key, "authsource") ||
+              (!mongoc_uri_option_is_bool(key) && !mongoc_uri_option_is_int32(key)
+               && !!strcasecmp(key, "readpreferencetags") &&
+               !!strcasecmp(key, "authmechanismproperties"));
+}
```

Comment:
Can you find a way to make this more readable? I suggest a couple of separate return statements, maybe temp variables. No need to squish all the logic into one expression.